packages/jest-cli/src/lib/update_global_config.js

```diff
@@ -7,16 +7,32 @@
  * @flow
  */
 
-import type {GlobalConfig, SnapshotUpdateState} from 'types/Config';
+import {replacePathSepForRegex} from 'jest-regex-util';
+
+import type {
+  GlobalConfig,
+  ReporterConfig,
+  SnapshotUpdateState,
+} from 'types/Config';
 
 type Options = {
+  bail?: boolean,
```

Comment:
Options is a subset of GlobalConfig. So we could use it as source of truth for our types by leveraging [`$PropertyType<T, k>`](https://flow.org/en/docs/types/utilities/#toc-propertytype) flow operator:
```jsx
type Options = {
  bail?: $PropertyType<GlobalConfig, 'bail'>,
  ...
}
```
(not mandatory for this PR though, just though you'd like to know about it)

Reply:
@thymikee Interesting! Not knowledgeable enough about Flow so far to have seen this one coming. I'm updating the `Options` type to ref `GlobalConfig` types whenever applicable (anything but `mode`, basically)